### ISSUE TITLE
python breakpoint in another module being ignored

### ISSUE THREAD

**Issue by pifalken** [emacs-lsp/dap-mode#494]
I'm trying to debug a large python codebase and have set a breakpoint in an external module. This is visible within the `dap-ui-breakpoints` and also in the breakpoints file, however it always gets skipped or ignored without any messages.

I've tried enabling `toggle-debug-on-error` and `toggle-debug-on-quit` yet get no hits.

`dap-print-io` shows nothing helpful either. The LSP server I'm using is pyright

```lisp
(dap-register-debug-template
  "Python :: Run file (buffer)"
  (list :type "python"
        :args ""
        :cwd nil
        :module nil
        :program nil
        :request "launch"
        :name "Python :: Run file (buffer)"))
```

**Comment by yyoncho** [comment]
I guess this - https://github.com/emacs-lsp/dap-mode/issues/184#issuecomment-570631667 ?

**Comment by pifalken** [comment]
@yyoncho But this is for ptvds right? I'm using debugpy

```lisp
      (use-package dap-mode
        :ensure t :after lsp-mode
        :config
        (require 'dap-python)
        (require 'dap-ui)
        (dap-mode t)
        (dap-ui-mode t)
        (dap-tooltip-mode t)
        (tooltip-mode t))
      (setq dap-python-debugger 'debugpy)
      ;(setq dap-auto-configure-features '(sessions locals controls tooltip))
      (setq dap-auto-configure-features '(locals controls tooltip))
```

**Comment by yyoncho** [comment]
> @yyoncho But this is for ptvds right? I'm using debugpy

Yes. Have you tested the config? AFAIK debugpy is fork of ptvsd so there is a great chance that it can work. If it doesn't work then we have to debug what vscode is sending to debugpy to enable external module debugging.

**Comment by pifalken** [comment]
> > @yyoncho But this is for ptvds right? I'm using debugpy
> 
> Yes. Have you tested the config? AFAIK debugpy is fork of ptvsd so there is a great chance that it can work. If it doesn't work then we have to debug what vscode is sending to debugpy to enable external module debugging.

That does in-fact work, should've tested earlier. Thank you.
I'm pretty new to dap-mode, is there a way to have this template already saved so I don't have to edit the template each time I close&reopen emacs?

**Comment by yyoncho** [comment]
You can use `dap-register-debug-template`

**Comment by yyoncho** [comment]
FTR I use dap-debug calls in `.templates.el` file and then I use M-x dap-debug-last to rerun it.

**Comment by yyoncho** [comment]
you can also store it in launch.json format
